lib/media/track/trackpublication.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const { EventEmitter } = require('events');
+const { buildLogLevels } = require('../../util');
+const { DEFAULT_LOG_LEVEL } = require('../../util/constants');
+const Log = require('../../util/log');
+let nInstances = 0;
+
+/**
+ * A {@link TrackPublication} represents a {@link Track} that
+ * has been published to a {@link Room}.
+ * @property {boolean} isTrackEnabled - whether the published {@link Track} is enabled
+ * @property {string} trackName - the {@link Track}'s name
+ * @property {Track.SID} trackSid - SID assigned to the published {@link Track}
+ * @emits TrackPublication#trackDisabled
+ * @emits TrackPublication#trackEnabled
+ */
+class TrackPublication extends EventEmitter {
+  constructor(isTrackEnabled, trackName, options) {
+    super();
+
+    options = Object.assign({
+      logLevel: DEFAULT_LOG_LEVEL
+    }, options);
+
+    const logLevels = buildLogLevels(options.logLevel);
+
+    Object.defineProperties(this, {
+      _instanceId: {
+        value: nInstances++
+      },
+      _isTrackEnabled: {
+        value: isTrackEnabled,
+        writable: true
+      },
+      _log: {
+        value: options.log || new Log('default', this, logLevels)
+      },
+      _trackSid: {
+        value: null,
+        writable: true
+      },
+      trackName: {
+        value: trackName
+      }
```

Comment:
This can be written as a getter.

Reply:
@markandrus What is the use of doing that. The name doesn't change, so initial assignment to the property should be fine.